src/sentry/auth/superuser.py

```diff
@@ -275,12 +295,56 @@ def set_logged_in(self, user, current_datetime=None):
         request = self.request
         if current_datetime is None:
             current_datetime = timezone.now()
+
+        token = get_random_string(12)
+
+        if not self._needs_validation():
+            self._set_logged_in(
+                expires=current_datetime + MAX_AGE,
+                token=token,
+                user=user,
+                current_datetime=current_datetime,
+            )
+
+            logger.info(
+                "superuser.logged-in",
+                extra={"ip_address": request.META["REMOTE_ADDR"], "user_id": user.id},
+            )
```

Comment:
Since these lines duplicate lines 341-350, I would pull the common code into a helper function. You could define the helper function right after `token = ...` above, nested inside `set_logged_in`, so that you can refer to variables in scope and not pass any arguments.